crates/core/src/store.rs

```diff
@@ -370,16 +364,22 @@ impl StoreBuilder {
 
         inner.limiter_async(move |data| &mut data.store_limits);
 
-        // With epoch interruption enabled, there must be _some_ deadline set
-        // or execution will trap immediately. Since this is a delta, we need
-        // to avoid overflow so we'll use 2^63 which is still "practically
-        // forever" for any plausible tick interval.
-        inner.set_epoch_deadline(u64::MAX / 2);
+        let delta = self.yield_interval.as_nanos() / self.epoch_tick_interval.as_nanos();
```

Comment:
This will panic if someone passes a `Duration` of 0 for the tick interval. We probably want to prevent that.

Reply:
Added `ensure!` above.